core/src/stored/bscan.cc

```diff
@@ -1090,28 +1090,14 @@ static bool CreateFileAttributesRecord(BareosDb* db,
 // For each Volume we see, we create a Medium record
 static bool CreateMediaRecord(BareosDb* db, MediaDbRecord* mr, Volume_Label* vl)
 {
-  struct date_time dt;
-  struct tm tm;
-
   // We mark Vols as Archive to keep them from being re-written
   bstrncpy(mr->VolStatus, "Archive", sizeof(mr->VolStatus));
   mr->VolRetention = 365 * 3600 * 24; /* 1 year */
   mr->Enabled = 1;
-  if (vl->VerNum >= 11) {
-    mr->set_first_written = true; /* Save FirstWritten during update_media */
-    mr->FirstWritten = BtimeToUtime(vl->write_btime);
-    mr->LabelDate = BtimeToUtime(vl->label_btime);
-  } else {
-    // DEPRECATED DO NOT USE
-    dt.julian_day_number = vl->write_date;
-    dt.julian_day_fraction = vl->write_time;
-    TmDecode(&dt, &tm);
-    mr->FirstWritten = mktime(&tm);
-    dt.julian_day_number = vl->label_date;
-    dt.julian_day_fraction = vl->label_time;
-    TmDecode(&dt, &tm);
-    mr->LabelDate = mktime(&tm);
-  }
+  ASSERT(vl->VerNum >= 11);
```

Comment:
A comment about this ASSERT might be useful.

Reply:
+1